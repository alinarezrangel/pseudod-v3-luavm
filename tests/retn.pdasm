PDVM 1.0
PLATFORM "pdcrt"

SECTION "code"
  ICONST 2001
  MK0CLZ 1
  DYNCALL 0, 0
  PRN
  NL
  ICONST 2003
  MK0CLZ 2
  DYNCALL 0, 0
  PRN
  NL
  ICONST 2004
  MK0CLZ 3
  DYNCALL 0, 0
  PRN
  NL

  ICONST 3001
  MK0CLZ 1
  DYNCALL 0, 1
  PRN
  NL
  PRN
  NL
  ICONST 3003
  MK0CLZ 2
  DYNCALL 0, 1
  PRN
  NL
  PRN
  NL
  ICONST 3004
  MK0CLZ 3
  DYNCALL 0, 1
  PRN
  NL
  PRN
  NL

  ICONST 4001
  MK0CLZ 1
  DYNCALL 0, 2
  PRN
  NL
  PRN
  NL
  PRN
  NL
  ICONST 4003
  MK0CLZ 2
  DYNCALL 0, 2
  PRN
  NL
  PRN
  NL
  PRN
  NL
  ICONST 4004
  MK0CLZ 3
  DYNCALL 0, 2
  PRN
  NL
  PRN
  NL
  PRN
  NL

  ICONST 5001
  MK0CLZ 1
  DYNCALL 0, 3
  PRN
  NL
  PRN
  NL
  PRN
  NL
  PRN
  NL
  ICONST 5003
  MK0CLZ 2
  DYNCALL 0, 3
  PRN
  NL
  PRN
  NL
  PRN
  NL
  PRN
  NL
  ICONST 5004
  MK0CLZ 3
  DYNCALL 0, 3
  PRN
  NL
  PRN
  NL
  PRN
  NL
  PRN
  NL
ENDSECTION

SECTION "procedures"
  PROC 1
    RETN 0
  ENDPROC
  PROC 2
    ICONST 20
    RETN 1
  ENDPROC
  PROC 3
    ICONST 1
    ICONST 2
    RETN 2
  ENDPROC
ENDSECTION

SECTION "constant pool"
  --
ENDSECTION

