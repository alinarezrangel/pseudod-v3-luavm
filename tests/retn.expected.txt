2001
2003
2004
NULO
3001
20
3003
1
3004
NULO
NULO
4001
20
NULO
4003
2
1
4004
NULO
NULO
NULO
5001
20
NULO
NULO
5003
2
1
NULO
5004
